src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+	}
+
+	public double ZoomLevel
+	{
+		get => (double)GetValue(ZoomLevelProperty);
+		set => SetValue(ZoomLevelProperty, value);
+	}
+
+	public double MinZoomLevel
+	{
+		get => (double)GetValue(MinZoomLevelProperty);
+		set => SetValue(MinZoomLevelProperty, value);
+	}
+
+	public double MaxZoomLevel
+	{
+		get => (double)GetValue(MaxZoomLevelProperty);
+		set => SetValue(MaxZoomLevelProperty, value);
+	}
+
+	public double HorizontalZoomCenter
+	{
+		get => (double)GetValue(HorizontalZoomCenterProperty);
+		set => SetValue(HorizontalZoomCenterProperty, value);
+	}
+
+	public double VerticalZoomCenter
+	{
+		get => (double)GetValue(VerticalZoomCenterProperty);
+		set => SetValue(VerticalZoomCenterProperty, value);
+	}
+
+	public double ScaleWheelRatio
+	{
+		get => (double)GetValue(ScaleWheelRatioProperty);
+		set => SetValue(ScaleWheelRatioProperty, value);
+	}
+
+	public double PanWheelRatio
+	{
+		get => (double)GetValue(PanWheelRatioProperty);
+		set => SetValue(PanWheelRatioProperty, value);
+	}
+
+	public bool IsPanAllowed
+	{
+		get => (bool)GetValue(IsPanAllowedProperty);
+		set => SetValue(IsPanAllowedProperty, value);
+	}
+
+	public double HorizontalOffset
+	{
+		get => (double)GetValue(HorizontalOffsetProperty);
+		set => SetValue(HorizontalOffsetProperty, value);
+	}
+
+	public double VerticalOffset
+	{
+		get => (double)GetValue(VerticalOffsetProperty);
+		set => SetValue(VerticalOffsetProperty, value);
+	}
+
+	public double VerticalMaxScroll
+	{
+		get => (double)GetValue(VerticalMaxScrollProperty);
+		set => SetValue(VerticalMaxScrollProperty, value);
+	}
+
+	public double VerticalMinScroll
+	{
+		get => (double)GetValue(VerticalMinScrollProperty);
+		set => SetValue(VerticalMinScrollProperty, value);
+	}
+
+	public double HorizontalMaxScroll
+	{
+		get => (double)GetValue(HorizontalMaxScrollProperty);
+		set => SetValue(HorizontalMaxScrollProperty, value);
+	}
+
+	public double HorizontalMinScroll
+	{
+		get => (double)GetValue(HorizontalMinScrollProperty);
+		set => SetValue(HorizontalMinScrollProperty, value);
+	}
+
+	public double HorizontalScrollValue
+	{
+		get => (double)GetValue(HorizontalScrollValueProperty);
+		set => SetValue(HorizontalScrollValueProperty, value);
+	}
+
+	public double VerticalScrollValue
+	{
+		get => (double)GetValue(VerticalScrollValueProperty);
+		set => SetValue(VerticalScrollValueProperty, value);
+	}
+
+	public double ViewPortHeight
+	{
+		get => (double)GetValue(ViewPortHeightProperty);
+		set => SetValue(ViewPortHeightProperty, value);
+	}
+
+	public double ViewPortWidth
+	{
+		get => (double)GetValue(ViewPortWidthProperty);
+		set => SetValue(ViewPortWidthProperty, value);
+	}
+
+	#endregion
+
+	private async Task RaiseRenderedContentUpdated()
+	{
+		await Task.Yield();
+		RenderedContentUpdated?.Invoke(this, EventArgs.Empty);
+	}
+
+	private void RegisterPropertyHandlers()
+	{
+		// Register for property changed events.
+		RegisterPropertyChangedCallback(ZoomLevelProperty, OnZoomLevelChanged);
```

Comment:
you could redirectly subscribe to the dp-changed in the dp registration
see other controls for example